core/src/main/java/com/sequenceiq/cloudbreak/service/StackCommonService.java

```diff
@@ -284,6 +313,15 @@ public FlowIdentifier retryInWorkspace(NameOrCrn nameOrCrn, Long workspaceId) {
         return retry.getFlowIdentifier();
     }
 
+    public void cancelInWorkspace(NameOrCrn nameOrCrn, Long workspaceId) {
+        Long stackId = stackService.getIdByNameOrCrnInWorkspace(nameOrCrn, workspaceId);
+
+        LOGGER.info("Cancelling existing flows for the stack : {}", stackId);
+        flowCancelService.cancelRunningFlows(stackId);
+        clusterService.updateClusterStatusByStackId(stackId, DetailedStackStatus.AVAILABLE, "fake update after cancelling the running flows");
+        stackUpdater.updateStackStatus(stackId, DetailedStackStatus.AVAILABLE, "fake update after cancelling the running flows");
```

Comment:
we cannot create such status reasons, also what happens if we cancel a flow and cluster is in bad state internally? (cp registration is not done, some ums call missing, etc). we cannot make stack available

Reply:
Related to the cancel API above, which is meant for development. If that goes, so does this.

Reply:
Fixed in the next push. Resolving conversation.